cli/medperf/entities/cube.py

```diff
@@ -122,9 +123,18 @@ def get(cls, cube_uid: str) -> "Cube":
 
         meta = comms.get_cube_metadata(cube_uid)
         cube = cls(meta)
-        cube.download()
-        cube.write()
-        return cube
+        attempt = 0
+        while attempt < config.cube_get_max_attempts:
+            logging.info(f"Downloading MLCube. Attempt {attempt + 1}")
+            cube.download()
+            if cube.is_valid():
+                cube.write()
+                return cube
+            attempt += 1
+        logging.error("Max download attempts reached")
+        cube_path = os.path.join(storage_path(config.cubes_storage), str(cube_uid + 1))
+        cleanup([cube_path])
+        raise RuntimeError("Could not successfully download the requested MLCube")
```

Comment:
I think we should keep medperf failures exiting with pretty errors rather than python exceptions. What do you think. (exception --> we have an internal bug)

Reply:
For now we can do that, but following from a previous discussion on another PR, I think entities should not call the UI directly. Using `pretty_error` restricts future implementations that might want to use medperf's entities and API from deciding when to interact with the UI and how to handle errors. What I think is the best approach for a future PR is to move all calls to `pretty_error` to the commands layer, and instead create our own exceptions that are called inside entities and utils. This way we're separating concerns, since entities no longer have to be aware of the UI.

Reply:
Let's talk about this in our discussion. I do agree that I should follow the way we're doing things now until we create a PR for custom exceptions.